Xamarin.PropertyEditing/ViewModels/BrushPropertyViewModel.cs

```diff
@@ -172,18 +172,42 @@ private void SetBrushType (CommonBrushType type)
 		{
 			StorePreviousBrush();
 
-			switch(type) {
-			case CommonBrushType.NoBrush:
-				Value = null;
-				break;
-			case CommonBrushType.Solid:
-				Value = Solid?.PreviousSolidBrush ?? new CommonSolidBrush (CommonColor.Black);
-				Solid?.CommitLastColor ();
-				Solid?.CommitHue ();
-				break;
-			case CommonBrushType.MaterialDesign:
-				MaterialDesign.SetToClosest ();
-				break;
+			switch (type) {
+				case CommonBrushType.MaterialDesign:
+					MaterialDesign.SetToClosest ();
+					break;
+
+				case CommonBrushType.NoBrush:
+					Value = null;
+					break;
+
+				case CommonBrushType.Resource:
+					if (Resource != null) {
+						Value = GetCommonBrushForResource (Resource);
```

Comment:
There's no explanation of why this change was made, nor any test covering  it.

Reply:
This was accidentally left in, while I was tracking down NoBrush issue. I'll remove it.